cli-plugins/manager/manager.go

```diff
@@ -0,0 +1,83 @@
+package manager
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/config"
+	"github.com/spf13/cobra"
+)
+
+// errPluginNotFound is the error returned when a plugin could not be found.
+type errPluginNotFound string
+
+func (e errPluginNotFound) NotFound() {}
+
+func (e errPluginNotFound) Error() string {
+	return "Error: No such CLI plugin: " + string(e)
+}
+
+type notFound interface{ NotFound() }
+
+// IsNotFound is true if the given error is due to a plugin not being found.
+func IsNotFound(err error) bool {
```

Comment:
My observation before I went this way was:
```
$ git grep func.IsNotFound
cli/manifest/store/store.go:func IsNotFound(err error) bool {
cli/registry/client/fetcher.go:func IsNotFound(err error) bool {
vendor/github.com/containerd/containerd/errdefs/errors.go:func IsNotFound(err error) bool {
vendor/github.com/docker/docker/errdefs/is.go:func IsNotFound(err error) bool {
vendor/k8s.io/apimachinery/pkg/api/errors/errors.go:func IsNotFound(err error) bool {
```
Consolidating them was a yakk I was hoping to avoid shaving in this already large PR ;-)

Reply:
Definitely not a showstopper; perhaps we should have a look at moving that package to its own repo, and use it across the board.